### PR TITLE
Envgateway supports prepend+append 

### PR DESCRIPTION
**Pull request by tobsan**
Some changes in filegateway were done as a consequence of this.

### CODE REVIEW

File: doc/chapters/gateways/index.rst [hunk 1]
```diff
@@ -271,10 +271,15 @@ a log warning.
 
 The configuration may also, optionally, specify the following parameters:
 
-- ``append`` (bool) If the environment variable is already defined by the gateway, append the new value to the value already defined. Defaults to false.
+- ``mode``: A string that can be either:
+    - ``set``: set the variable (this is the default, so one does not usually need to set it)
+    - ``append``: append the value given to the previous value for the given variable.
+    - ``prepend``: prepend the value given to the previous value for the given variable.
+- ``separator``: A string that will be squeezed in between the old and new value if one wants to
```
Comment (breakreturn) [inline doc/chapters/gateways/index.rst:278]: Squeezed? Could use more boring but "correct" language :)

File: doc/chapters/gateways/index.rst [hunk 2]
```diff
@@ -297,8 +302,9 @@ There is also the possibility to append to an already defined variable::
     [
         {
             "name": "SOME_ENVIRONMENT_VARIABLE",
-            "value": ":/some/path",
-            "append": true
+            "value": "/some/path",
+            "mode": "append"
```
Comment (DunderRoffe) [inline doc/chapters/gateways/index.rst:306]: Make "append" big letters or non case sensitive

**Review by DunderRoffe — CHANGES_REQUESTED** [review]

File: doc/chapters/gateways/index.rst (hunk 2 above)
Comment (tobsan) [reply doc/chapters/gateways/index.rst:306]: Making it case insensitive seems reasonable, I'm doing that.

**Review by DunderRoffe — APPROVED** [review]
Awesomesauce

**Pull request merged**